ipn/ipnlocal/local.go

```diff
@@ -453,6 +453,24 @@ func (b *LocalBackend) updateStatus(sb *ipnstate.StatusBuilder, extraLocked func
 	}
 }
 
+// exitNodeIP returns the exit node IP from prefs.
+func (b *LocalBackend) exitNodeIP() netaddr.IP {
+	if !b.prefs.ExitNodeIP.IsZero() {
```

Comment:
b.prefs is guarded by b.mu.  So either you need to acquire b.mu (and make sure that's safe for callers) or rename this `exitNodeIPLocked` (the Locked suffix conventionally means the mutex must already be held by the caller)

Also, check that b.prefs is non-nil.